### PR TITLE
chore: connect HGI UI to its API [DET-4638]

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/api-ts-sdk/Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ all: get-deps build
 .PHONY: get-deps-swagger
 get-deps-swagger:
 	mkdir -p deps
-	curl https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.14/swagger-codegen-cli-2.4.14.jar \
+	[ -f deps/swagger-codegen-cli.jar ] || curl https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.14/swagger-codegen-cli-2.4.14.jar \
```
Comment (hkang1) [inline webui/api-ts-sdk/Makefile:10]: nice, no need to re-download this every time. Just need to run `make clean` if we ever need to get a new version which shouldn't happen too often.

File: webui/react/src/components/Json.tsx [hunk 2]
```diff
@@ -27,17 +27,19 @@ const row = (
...
   </li>;
 };
 
-const Json: React.FC<Props> = ({ json, transformLabel }: Props) => {
+const Json: React.FC<Props> = ({ json, translateLabel }: Props) => {
 
   return (
     <ul className={css.base}>
-      {Object.entries(json).map(([ label, value ]) => row(label, value, transformLabel))}
+      {Object.entries(json).map(([ label, value ]) => row(label, value, translateLabel))}
```
Comment (hkang1) [inline webui/react/src/components/Json.tsx:42]: **suggestion:** `row` should probably just be a react function component `JsonRow` inside this same file.

File: webui/react/src/components/ResourcePoolCard.tsx [hunk 3]
```diff
@@ -16,18 +17,21 @@ import ResourcePoolDetails from './ResourcePoolDetails';
...
-    case 'aws':
+    case V1ResourcePoolType.AWS:
       iconSrc = awsLogo;
       break;
-    case 'gcp':
+    case V1ResourcePoolType.GCP:
       iconSrc = gcpLogo;
       break;
-    case 'static':
+    case V1ResourcePoolType.K8S: // TODO add dedicated logo
```
Comment (hkang1) [inline webui/react/src/components/ResourcePoolCard.tsx:32]: I'll get this over to you soon.

File: webui/react/src/components/ResourcePoolCard.tsx [hunk 4]
```diff
@@ -88,6 +94,9 @@ const ResourcePoolCard: React.FC<Props> = ({ containerStates, resourcePool: rp }
     [],
   );
 
+  const descriptionClasses = [ css.description ];
+  if (!description) descriptionClasses.push(css.empty);
+
```
Comment (hkang1) [inline webui/react/src/components/ResourcePoolCard.tsx:99]: **nit:** I've been mostly having these class definitions near the top of the component and hoisting all the variable definitions near the top of the function.

File: webui/react/src/services/apiConfig.ts [hunk 5]
```diff
@@ -34,7 +33,7 @@ export const detApi = {
   Cluster: new Api.ClusterApi(ApiConfig),
   Commands: new Api.CommandsApi(ApiConfig),
   Experiments: new Api.ExperimentsApi(ApiConfig),
-  InternalApi: new Api.InternalApi(ApiConfig),
+  Internal: new Api.InternalApi(ApiConfig),
```
Comment (hkang1) [inline webui/react/src/services/apiConfig.ts:36]: 👍

File: webui/react/src/services/decoder.ts [hunk 6]
```diff
@@ -41,6 +41,12 @@ export const jsonToDeterminedInfo = (data: Sdk.V1GetMasterResponse): types.Deter
   };
 };
 
+export const mapV1ResourcePool = (
```
Comment (hkang1) [inline webui/react/src/services/decoder.ts:44]: **nit:** we should probably come to a naming convention for these types of function. We have `decode` prefixes, and XToY, etc. I like `map` it's pretty succinct. How about something like `mapSdkToResourcePool`?

File: webui/react/src/themes.ts [hunk 7]
```diff
@@ -153,7 +153,7 @@ export const lightTheme: Theme = {
     states: {
       active: '#009bde',
       failed: '#cc0000',
-      free: '#f7f7f7',
+      free: '#ececec',
```
Comment (hkang1) [inline webui/react/src/themes.ts:156]: **nit:** how about going white instead now that the bg is darker? #ececec makes the wording a bit harder to read

**Review by hkang1 — APPROVED** [review]
Great work with this PR! Wrote a bunch of small things but nothing blocking.

File: webui/api-ts-sdk/Makefile (hunk 1 above)
Comment (hamidzr) [reply webui/api-ts-sdk/Makefile:10]: > Just need to run make clean if we ever need to get a new version which shouldn't happen too often.

exactly. I ended up addressing this in a better way covering this case as well and put up a separate PR https://github.com/determined-ai/determined/pull/1838  so I'll take it out of here.

File: webui/react/src/services/decoder.ts (hunk 6 above)
Comment (hamidzr) [reply webui/react/src/services/decoder.ts:44]: yea I agree they're all over the place

about mapSdkToResourcePool : sdk sounds short but I think these are more bindings and sdk might not be very accurate can you think of something as short that would be closer to binding?

File: webui/react/src/themes.ts (hunk 7 above)
Comment (hamidzr) [reply webui/react/src/themes.ts:156]: hmm the bg isn't non-white everywhere eg the resourcepool card. I'd have to style it differently if we wanna go with white. I can make it a bit bright

File: webui/react/src/components/ResourcePoolCard.tsx (hunk 3 above)
Comment (hamidzr) [reply webui/react/src/components/ResourcePoolCard.tsx]: thanks, I added it in.

**Pull request merged**